game/infra/profile.lua

```diff
@@ -21,43 +22,43 @@ local filesystem = love.filesystem
 
 -- LOCALS --
 local PROFILE = {}
-local __COMPRESS__ = false
 local _id_generator
 local _metadata
-
-local function _compress(str) --> str
-  if not __COMPRESS__ then return str end
-  return assert(ZIP.compress(str:gsub(" ", ""), "lz4", 9))
-end
+local _savethread
+local _channel
+local _compress
 
 local function _decompress(str) --> str
-  if not __COMPRESS__ then return str end
+  if not _compress then return str end
   return assert(ZIP.decompress(str, "lz4"))
 end
 
-local function _encode(t) --> str
-  return _compress(JSON.encode(t, {indent = true}))
-end
-
 local function _decode(str) --> table
   return JSON.decode(_decompress(str))
 end
 
 local function _deleteInput()
-  INPUT.delete(CONTROL_PATH)
+  return filesystem.remove(CONTROL_PATH)
 end
 
 local function _loadInput()
-  -- setup input
-  local loaded_input = INPUT.load(CONTROL_PATH, _decode)
-  if not loaded_input then
-    local inputmap = DB.loadSetting('controls')
-    INPUT.setup(inputmap)
+  local inputmap
+  if filesystem.getInfo(CONTROL_PATH, { type = 'file' }) then
+    local filedata = assert(filesystem.newFileData(CONTROL_PATH))
+    inputmap = _decode(filedata:getString())
+  else
+    inputmap = DB.loadSetting('controls')
   end
+  return INPUT.setup(inputmap)
```

Comment:
What are these changes about?

Reply:
the input module was writing the controls save, but that is an optional feature. i just stopped using it and let the writing thread deal with all writing to file